src/hooks/useXplorersAI.ts

```diff
@@ -0,0 +1,69 @@
+"use client";
+
+import { callAzureOpenAI } from "@/app/api/chat/ai";
+import { useState, useRef, ChangeEvent, MouseEvent } from "react";
+export interface FileUploadEvent extends ChangeEvent<HTMLInputElement> {
+  target: HTMLInputElement;
+}
+export interface ChatMessage {
+  role: "user" | "system";
+  content: string;
+}
+export const useXplorersAI = () => {
+  const [value, setValue] = useState<string>("");
+  const textareaRef = useRef<HTMLTextAreaElement | null>(null);
+  const [messages, setMessages] = useState<ChatMessage[]>([
+    { role: "user", content: "" },
+  ]);
+  const handleInputChange = (e: ChangeEvent<HTMLTextAreaElement>) => {
+    const textValue = e.target.value;
+    setValue(textValue);
+
+    if (textareaRef.current) {
+      textareaRef.current.style.height = "auto";
+      textareaRef.current.style.height = `${textareaRef.current.scrollHeight}px`;
+    }
+  };
+
+  const handleFileUpload = (e: FileUploadEvent) => {
+    const file = e.target.files ? e.target.files[0] : null;
+    if (file) {
+      console.log("File uploaded:", file);
+    }
+  };
```

Comment:
Are we doing anything with file upload ?

Reply:
Not now